src/pkg/packager/create.go

```diff
@@ -59,6 +61,13 @@ func (p *Packager) Create(baseDir string) error {
 		p.cfg.IsInitConfig = true
 	}
 
+	// Before we compose the components (and render the imported OCI components), we need to remove any components that are not needed for a differential build
+	if p.cfg.CreateOpts.DifferentialData.DifferentialPackagePath != "" {
+		if err := p.removeDifferentialComponentsFromPackage(); err != nil {
+			return err
+		}
+	}
```

Comment:
Same here - the length check in `removeDifferentialComponentsFromPackage` will have the same effect as this `nil` check